src/traces/histogram/calc.js

```diff
@@ -43,9 +43,11 @@ module.exports = function calc(gd, trace) {
     var pos0 = pa.makeCalcdata(trace, maindata);
 
     // calculate the bins
-    var binAttr = maindata + 'bins',
-        binspec;
-    if((trace['autobin' + maindata] !== false) || !(binAttr in trace)) {
+    var binAttr = maindata + 'bins';
+    var autoBinAttr = 'autobin' + maindata;
+    var binspec = trace[binAttr];
+    if((trace[autoBinAttr] !== false) || !binspec ||
+            binspec.start === null || binspec.end === null) {
```

Comment:
note that [`cleanBins`](https://github.com/plotly/plotly.js/blob/bf1aa6ba015c304c8cf07f4027637ac2c23addd9/src/traces/histogram/calc.js#L40) provides a `size` even if there are no `start` or `end`. This also doesn't allow you to specify a `start` and no `end` or things like that. So *partial* bin definition might do strange things... but that I would definitely consider a feature we could propose to add, rather than a bug to fix.

Also note that `!(binAttr in trace)` in the old version never failed, since we default to `null` for `start` and `end` (which might have been important to enable `cleanBins`?) - so in principle I could omit `!binspec` here, just thought it prudent in case there's some context I'm not aware of that can call this `calc`.